src/effects/VST3/internal/PlugFrame.cpp

```diff
@@ -23,26 +24,32 @@ internal::PlugFrame::~PlugFrame()
    FUNKNOWN_DTOR;
 }
 
-static void UpdateWindowSize(wxWindow* window, const wxSize& newSize, bool fixed)
+void internal::PlugFrame::init(Steinberg::IPlugView* view, Steinberg::ViewRect* size)
 {
-   if(fixed)
-   {
-      //Update min/max if plugin window has fixed size
-      //but for some reason resize was requested
-      window->SetMinSize(newSize);
-      window->SetMaxSize(newSize);
-   }
-   window->SetSize(newSize);
+   if(!mInitialized)
+      resizeView(view, size);
 }
 
 Steinberg::tresult internal::PlugFrame::resizeView(Steinberg::IPlugView* view, Steinberg::ViewRect* newSize)
 {
    if(auto window = mWindow.get())
    {
-      auto fixed = view->canResize() != Steinberg::kResultTrue;
-      UpdateWindowSize(window, {newSize->getWidth(), newSize->getHeight()}, fixed);
+      auto topWindow = wxGetTopLevelParent(window);
+
+      wxWindowUpdateLocker windowUpdateLocker(topWindow);
+
+      window->SetInitialSize({newSize->getWidth(), newSize->getHeight() });
```

Comment:
A deleted comment said
```
           //No need to accommodate the off-by-one error with Steinberg::ViewRect
           //as we do it with wxRect
```

Are you sure there are no implications of off-by-one errors here in the rewrite?

Reply:
Where is that error?